src/Frontend/Core/Engine/Block/Extra.php

```diff
@@ -163,23 +163,36 @@ public function execute()
     public function getAction()
     {
         // no action specified?
-        if ($this->action === null) {
+        if (null === $this->action) {
             // get first parameter
             $actionParameter = $this->URL->getParameter(0);
 
-            // unknown action and not provided in URL
-            if ($actionParameter === null) {
-                $this->setAction($this->config->getDefaultAction());
-            } else {
+            // set default action
+            $this->setAction($this->config->getDefaultAction());
+
+            // if we have module routing lets check it first and set action parameter
+            if ($this->config->hasRouter()) {
+                try {
+                    $parameters = $this->config->getRouter()->match('/' . implode('/', $this->URL->getParameters()));
+                } catch (\Exception $e) {
+                    $parameters = array();
+                }
+
+                $actionParameter = isset($parameters['_action'])?$parameters['_action']:null;
```

Comment:
This doesn't follow psr2 coding styles. You should have spaces around the '?' sign and the ':'.